src/index.ts

```diff
@@ -46,9 +49,38 @@ export class PromiseQueue {
 
 		this.concurrency = concurrency;
 		this.maxSize = maxSize;
-		this.maxAge = maxAge;
 		this.order = order;
 		this.next = order === 'lifo' ? 'pop' : 'shift';
+
+		if (maxAge > 0 && maxAge < Infinity) {
+			const maxAgeSeconds = Math.floor(maxAge / 1000);
+			const maxAgeNanoSeconds = (maxAge - maxAgeSeconds) * 1000000;
```

Comment:
I think it's reasonable to ignore nanoseconds and just use the *ceil*'d value. It'll also simplify the conditionals below.

Reply:
The reason I included nanoseconds is not because I care about nanosecond resolution (it's not even able to be specified) but because of the millisecond resolution because otherwise you only get resolution to the second, that said the interval timer limits the resolution as well so might as well go for the full performance